TypeCobol/Compiler/Directives/IBMCompilerOptions.cs

```diff
@@ -717,6 +820,27 @@ public bool IsOptionDeprecated(string optionWord, [MaybeNullWhen(false)] out str
         {
             return deprecatedOptions.TryGetValue(optionWord, out warningMessage);
         }
+
+        /// <summary>
+        /// If optionWord is not an allowed option this method returns the corresponding error message to be displayed
+        /// </summary>
+        public bool IsOptionNotAllowed(string optionWord, [MaybeNullWhen(false)] out string errorMessage)
+        {
+            return notAllowedOptions.TryGetValue(optionWord, out errorMessage);
+        }
+
+        /// <summary>
+        /// Get the option name from an option word (which can be an abbreviation or a negation)
+        /// </summary>
+        public string? GetOptionName(string optionWord)
```

Comment:
Return type should be `IBMCompilerOptionName?` instead of `string?`.

Reply:
Also this could be turned into the TryGet pattern for clients which are interested in testing only the existence of a name

Reply:
Return type changed to `IBMCompilerOptionName?`

Reply:
Done + optionWordToOptionName dictionary created with OrdinalIgnoreCase to avoid ToUpper